core/src/main/java/io/onetable/iceberg/IcebergClient.java

```diff
@@ -206,6 +210,33 @@ public Optional<OneTableMetadata> getTableMetadata() {
     if (table == null) {
       return Optional.empty();
     }
+    rollbackCorruptCommits();
```

Comment:
@ashvina I've moved this to the class constructor for now to avoid more methods in the target interface. if there are other use cases we can add an interface